### PR TITLE
STJS-586 sentry

### PR DESCRIPTION
**Pull request by wirus15**

### CODE REVIEW

File: src/client/config/ConfigResolver.ts [hunk 1]
```diff
@@ -50,7 +50,8 @@ export class ConfigResolver {
       submitOnSuccess: this._getValueOrDefault(config.submitOnSuccess, DefaultConfig.submitOnSuccess),
       successCallback: this._getValueOrDefault(config.successCallback, DefaultConfig.successCallback),
       translations: this._getValueOrDefault(config.translations, DefaultConfig.translations),
-      visaCheckout: this._setVisaCheckoutConfig(config.visaCheckout, DefaultConfig.visaCheckout)
+      visaCheckout: this._setVisaCheckoutConfig(config.visaCheckout, DefaultConfig.visaCheckout),
+      sentryDsn: this._getValueOrDefault(config.sentryDsn, '')
```
Comment (pgs-gmartowski) [inline src/client/config/ConfigResolver.ts:54]: I think this should be in alphabetical order, if we had such convention before

Comment (pgs-gmartowski) [reply src/client/config/ConfigResolver.ts:54]: let's put default value of `sentryDsn` into `DefaultConfig` as well

File: src/shared/services/sentry/EventScrubber.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Event } from '@sentry/types';
+import { EventScrubber } from './EventScrubber';
+
+describe('EventScrubber', () => {
```
Comment (pgs-gmartowski) [inline src/shared/services/sentry/EventScrubber.spec.ts:4]: I was using `//given // when // then` convention before; should we continue this or leave it ?

File: src/shared/services/sentry/EventScrubber.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        config: {
+          jwt: 'some-long-jwt-value',
+          foo: 'bar'
+        }
+      }
+    };
+
+    const result = eventScrubber.scrub(event);
+
+    expect(result.extra.config).toEqual({
```
Comment (pgs-gmartowski) [inline src/shared/services/sentry/EventScrubber.spec.ts:23]: you're using only `config` property here, so I would do sth like:
`
    const { config }= eventScrubber.scrub(event).extra;

    expect(config).toEqual({
`
but it's nth :)

File: src/shared/services/sentry/EventScrubber.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const event: Event = {
+      request: {
+        url: 'https://webservices.securetrading.net?foo=bar&jwt=some-long-jwt&xyz=abc',
+        query_string: 'jwt=some-long-jwt'
+      }
+    };
+
+    const result = eventScrubber.scrub(event);
+
+    expect(result.request.url).toBe('https://webservices.securetrading.net?foo=bar&jwt=*****&xyz=abc');
```
Comment (pgs-gmartowski) [inline src/shared/services/sentry/EventScrubber.spec.ts:39]: looks like `https://webservices.securetrading.net?foo=bar` is repeated twice; maybe it's good to use template strings here

**Comment by wirus15** [comment]
@pgs-gmartowski I made changes from your review. Plus I moved the sentry dsn to environment and added `errorReporting` property to config, like @james-simpson-ST suggested.

File: example/json/config.json [hunk 5]
```diff
@@ -59,6 +59,7 @@
   "deferInit": false,
   "disableNotification": false,
   "errorCallback": null,
+  "errorReporting": false,
```
Comment (pgs-gmartowski) [inline example/json/config.json:62]: I think James suggested that it should be named `debug`

File: src/shared/services/sentry/SentryService.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    this.sentry.setTag('hostName', this.sentryContext.getHostName());
+    this.sentry.setTag('frameName', this.sentryContext.getFrameName());
+
+    this.sentry.init({
+      dsn,
+      environment: this.sentryContext.getEnvironmentName(),
+      release: this.sentryContext.getReleaseVersion(),
+      whitelistUrls: ['https://webservices.securetrading.net'],
```
Comment (pgs-gmartowski) [inline src/shared/services/sentry/SentryService.ts:29]: just wondering if such constant values should be somewhere outside of class, or inside as static properties ?
What should be our approach ?

Comment (wirus15) [reply src/shared/services/sentry/SentryService.ts:29]: good point 👍

File: example/json/config.json (hunk 5 above)
Comment (wirus15) [reply example/json/config.json:62]: I remember he suggested `debug` but I thought `errorReporting` better describes what this property does. We could use `debug` to put more verbose `console.logs`.

**Pull request merged**